packages/cli/src/commands/setup.ts

```diff
@@ -0,0 +1,187 @@
+
+          const isExecExists = await pathExists(this.nextExecPath);
+          if (isExecExists) {
+            return 'Version already installed';
+          }
+
+          const hostDist = `${process.platform.replace('32', '')}-${process.arch}`;
+          const archiveExt = !stdenv.isWindows ? 'tar.gz' : 'zip';
+          const remoteArchive = await configuFilesApi({
+            method: 'GET',
+            url: `/cli/versions/${version}/configu-v${version}-${hostDist}.${archiveExt}`,
+            responseType: 'stream',
+          });
+          const archivePath = path.join(binDir, `configu.${archiveExt}`);
+          const localArchive = await fs.open(archivePath);
+          const writer = localArchive.createWriteStream();
+          remoteArchive.data.pipe(writer);
+          await stream.finished(writer);
+          if (archiveExt === 'tar.gz') {
+            await tar.x({ file: archivePath, gzip: true, cwd: binDir });
+          } else {
+            const zip = new Zip(archivePath);
+            zip.extractAllTo(binDir, true);
+          }
+          return 'Version installed';
+        },
+      },
+      {
+        enabled: isGlobalOrVersionFlag,
+        title: 'Creating bin shims',
+        // https://github.com/pnpm/pnpm/blob/main/packages/plugin-commands-setup/src/setup.ts#L67
+        task: async (message) => {
+          // windows can also use shell script via mingw or cygwin so no filter
+          const shellScript = ['#!/bin/sh', `exec ${this.nextExecPath} "$@"`].join('\n');
+          await fs.writeFile(path.join(this.context.paths.home, 'configu'), shellScript, { mode: 0o755 });
+
+          if (stdenv.isWindows) {
+            const batchScript = ['@echo off', `${this.nextExecPath} %*`].join('\n');
+            await fs.writeFile(path.join(this.context.paths.home, 'configu.cmd'), batchScript, { mode: 0o755 });
+
+            const powershellScript = `${this.nextExecPath} $args`;
+            await fs.writeFile(path.join(this.context.paths.home, 'configu.ps1'), powershellScript, { mode: 0o755 });
+          }
+          return 'Bin shims created';
+        },
+      },
+      {
+        enabled: isGlobalOrVersionFlag,
+        title: 'Adding bin to PATH',
+        task: async (message) => {
+          const binEnvPath = await addDirToEnvPath(this.context.paths.home, {
+            position: 'start',
+            proxyVarName: 'CONFIGU_HOME',
+            overwrite: true,
+            configSectionName: 'configu',
+          });
+          debug('addDirToEnvPath', binEnvPath);
+          if (binEnvPath.configFile?.changeType !== 'skipped') {
+            output.push(`Open a new terminal to apply the changes.`);
+          }
+          return 'Bin added to PATH';
+        },
+      },
+    ]);
+
+    if (isGlobalOrVersionFlag) {
+      output.push(`Run \`configu --help\` to see the available commands.`);
+      output.push(`Run \`configu init\` to get started.`);
```

Comment:
The init command currently doesn't exist, maybe we should omit this until it returns to prevent confusing the user.